client/login/wp-login/test/test.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External Dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal Dependencies
+ */
+import { Login } from '../index.jsx';
+import useFakeDom from 'test/helpers/use-fake-dom';
+
+const defaultProps = {
+	magicLoginEnabled: true,
+	magicLoginView: '',
+	translate: ( str ) => {
+		return str;
+	}
+};
+
+describe( 'Login', () => {
+	useFakeDom();
+
+	describe( 'footerLinks', () => {
+		it( 'should not have a return button before the state is loaded', () => {
+			const login = new Login( defaultProps );
+			const links = login.footerLinks();
+
+			expect( links ).to.have.length( 2 );
+		} );
+
+		it( 'should not have a return button after the state is loaded, but there is no history', () => {
+			const login = new Login( defaultProps );
+			login.state.loaded = true;
```

Comment:
I would rather recommend using [shallow rendering](https://github.com/airbnb/enzyme/blob/master/docs/api/shallow.md) for testing React components :)

Example here: https://github.com/Automattic/wp-calypso/blob/13c51873c56898ebd2dad72564d97822de4e5a22/client/components/notice/test/index.js

This test also uses `identity` to stub `translate` property.